python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -186,8 +191,16 @@ def from_assets(
                 auto_materialize_policies_by_key.update(asset.auto_materialize_policies_by_key)
                 backfill_policies_by_key.update({key: asset.backfill_policy for key in asset.keys})
                 if len(asset.keys) > 1 and not asset.can_subset:
+                    all_required_keys = {*asset.check_keys, *asset.keys}
                     for key in asset.keys:
                         required_multi_asset_sets_by_key[key] = asset.keys
+                    for key in all_required_keys:
+                        required_assets_and_checks_by_key[key] = all_required_keys
+                elif len(asset.keys) == 1 and asset.check_specs:
+                    required_keys = {asset.key, *asset.check_keys}
+                    for key in required_keys:
```

Comment:
note that if we have an asset with 1000 checks, required_assets_and_checks_by_key gets 1000 entries each pointing at required_keys. I think that's fine, they all point at the same underlying set.

The alternative would be to change access to `get_required_asset_checks_for_asset_key(`. That would work with current api's but not if we add @asset_multi_check.